### PR TITLE
14.0 mig l10n fi business code validate

### PR DESCRIPTION
**Pull request by jarmokortetjarvi**
Migrated business code validation from `12.0` to `14.0`. 
Module version `13.0` is missing, but this could be back-ported to `13.0` just by changing the version number.

Actual diff
https://github.com/OCA/l10n-finland/compare/12.0...jarmokortetjarvi:14.0-mig-l10n_fi_business_code_validate?expand=1

Only functional change for the earlier version is the addition of `country_id` to the constraint. 
In earlier versions you can have an incorrect business code if you set it before setting the country.

The diff looks large as the validation was in partner category, which was dropped when the business code field was simplified into a Char-field.

### CODE REVIEW

**Review by aisopuro — APPROVED** [review]
LGTM

File: l10n_fi_business_code_validate/README.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+==============================
+Partner business code validate
+==============================
+
+Adds a business code validation and format checks for business id
+
+Supported countries:
+
+- Finland
+   - https://www.vero.fi/contentassets/38c6e70f10b342cdba4716e904c3edbf/tarkistusmerkin-laskenta.pdf
```
Comment (mlaitinen) [inline l10n_fi_business_code_validate/README.rst:14]: This seems to be a dead link

File: l10n_fi_business_code_validate/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Adds a business code (business ID) validation and format check for business id
+
+Supported countries:
+
+- Finland
+   - https://www.vero.fi/globalassets/tietoa-verohallinnosta/ohjelmistokehittajille/yritys--ja-yhteis%C3%B6tunnuksen-ja-henkil%C3%B6tunnuksen-tarkistusmerkin-tarkistuslaskenta.pdf
```
Comment (mlaitinen) [inline l10n_fi_business_code_validate/readme/DESCRIPTION.rst:6]: This link seems to work just fine. Maybe drop the `README.rst` from the PR? Me thinks OCA bot should automatically generate it based on the `readme/` files.

**Review by mlaitinen — APPROVED** [review]
[OCA bot](https://github.com/OCA/oca-github-bot#id4) will regenerate `README.rst` when pushing to a main branch, so I'll approve this despite of the dead link.

**Pull request merged**